features/volgindicaties.feature

```diff
@@ -0,0 +1,154 @@
+      En gebruiker "B" heeft een volgindicatie voor de persoon met burgerservicenummer "999992351" en einddatum "2021-03-01"
+      En geen enkele andere gebruiker heeft een volgindicatie voor de persoon met burgerservicenummer "999992351"
+      Als gebruiker "A" op 5 maart 2021 een volgindicatie zet op de persoon met burgerservicenummer "999992351" met einddatum "2021-03-05"
+      Dan stuurt de applicatie een Av01 bericht met burgerservicenummer "999992351" naar het GBA-V
+
+   Scenario: Een gebruiker beëindigt de volgindicatie voor een persoon die ook door een andere gebruiker gevolgd wordt
+      Gegeven de persoon met burgerservicenummer "999992971" is NIET ingeschreven in deze gemeente
+      En gebruiker "A" heeft een volgindicatie voor de persoon met burgerservicenummer "999992971" en einddatum "2021-03-15"
+      En gebruiker "B" heeft een volgindicatie voor de persoon met burgerservicenummer "999992971" en einddatum "2021-06-30"
+      En geen enkele andere gebruiker heeft een volgindicatie voor de persoon met burgerservicenummer "999992971"
+      Als gebruiker "A" op 5 maart 2021 een volgindicatie zet op de persoon met burgerservicenummer "999992971" met einddatum "2021-03-05"
+      Dan stuurt de applicatie GEEN Av01 bericht met burgerservicenummer "999992971" naar het GBA-V
+
+   Scenario: Een gebruiker heropent een beëindigde volgindicatie
+      Gegeven de persoon met burgerservicenummer "000009945" is NIET ingeschreven in deze gemeente
+      En gebruiker "A" heeft een volgindicatie voor de persoon met burgerservicenummer "000009945" en einddatum "2021-03-01"
+      En geen enkele andere gebruiker heeft een volgindicatie voor de persoon met burgerservicenummer "000009945"
+      Als gebruiker "A" op 5 maart 2021 een volgindicatie zet op de persoon met burgerservicenummer "000009945" en geen einddatum
+      Dan stuurt de applicatie een Ap01 bericht met burgerservicenummer "000009945" naar het GBA-V
+
+   Scenario: Een gebruiker heropent een beëindigde volgindicatie die eerder door een andere gebruiker gevolgd werd
+      Gegeven de persoon met burgerservicenummer "999990949" is NIET ingeschreven in deze gemeente
+      En gebruiker "A" heeft een volgindicatie voor de persoon met burgerservicenummer "999990949" en einddatum "2021-03-01"
+      En geen enkele andere gebruiker heeft een volgindicatie voor de persoon met burgerservicenummer "999990949"
+      En de applicatie heeft op 1 maart 2021 een Av01 bericht met burgerservicenummer "999990949" naar het GBA-V gestuurd
+      Als gebruiker "B" op 5 maart 2021 een volgindicatie zet op de persoon met burgerservicenummer "999990949" en geen einddatum
+      Dan stuurt de applicatie een Ap01 bericht met burgerservicenummer "999990949" naar het GBA-V
+
+   Scenario: De einddatum van een volgindicatie is vandaag voor een persoon die ook door een andere gebruiker gevolgd wordt
```

Comment:
rare scenario. In de Als stap verwacht ik een actie en hier wordt alleen aangegeven wat de datum van vandaag is. Geen actie is geen reactie, dus dan geldt de Dan stap altijd

Reply:
het gaat hier niet om een externe gebeurtenis of handeling. De applicatie moet dadelijks controleren of er volgindicaties bij GBA-V actief zijn waar geen enkele gebruiker meer een actieve volgindicatie op heeft. Van buitenaf gebeurt er niks, behalve dat er een bepaalde datum is bereikt.
Hoe zou jij dit omschrijven in de feature? De datum in Gegeven opnemen i.p.v. in Als?
Of is het beter niet echte datums te noemen maar "gisteren", "vandaag" en "morgen"?

Reply:
Naar mijn mening stuurt de scenario's naar een bepaalde implementatie, i.p.v. het beschrijven van het gewenste gedrag van de Update API. Zoals de implementatie nu is beschreven, wordt verwacht dat een Av01 bericht wordt gestuurd zodra er geen actieve volgindicatie meer is. Maar is dit wel handig? Want als een consumer op een later moment zijn volgindicatie heractiveert met een nieuwe toekomstige einddatum, en er zijn wijzigingen in de periode tussen het sturen van het Av01 bericht en het heractiveren, dan krijgt de consumer deze wijzigingen niet omdat de Update API deze persoon in die periode niet heeft gevolgd.

Reply:
Los hiervan, zoals de scenario is beschreven, kan het niet worden gebruikt om geautomatiseerde testen van te maken

Reply:
> Naar mijn mening stuurt de scenario's naar een bepaalde implementatie, i.p.v. het beschrijven van het gewenste gedrag van de Update API. (...) Maar is dit wel handig? Want als een consumer op een later moment zijn volgindicatie heractiveert met een nieuwe toekomstige einddatum, en er zijn wijzigingen in de periode tussen het sturen van het Av01 bericht en het heractiveren, dan krijgt de consumer deze wijzigingen niet omdat de Update API deze persoon in die periode niet heeft gevolgd.

Dit is toch precies de beschrijving van het gewenste gedrag? ls we als gewenst gedrag een "afkoelperiode" hadden gewild, had dat in het scenario gestaan. Zo'n businesseis heb ik niet, wel de eis dat we alleen personen volgen in GBA-V waar ook daadwerkelijk een volgindicatie op staat.

Reply:
Het is niet verplicht dat een scenario altijd een `Als` stap moet hebben. Ik zou de `Als` stap als een `Gegeven` stap opnemen, dus:

En de datum vandaag is 1 maart 2021

En de Dan stap wijzigen naar

Dan is er door de update applicatie geen Av01 bericht met burgerservicenummer "999990962" naar het GBA-V gestuurd